pkg/controller/subnetcache.go

```diff
@@ -0,0 +1,163 @@
+	subnetDn := obj.GetAttrStr("dn")
+	subnetIp := obj.GetAttrStr("ip")
+	
+	cont.log.Debug("SubnetChanged for dn: ", subnetDn)
+	if _, ok := cont.apicConn.CachedSubnetDns[subnetDn]; !ok {
+		cont.UpdateSubnetDnCache(subnetDn, subnetIp, aciVrfDn)
+		cont.scheduleRdConfig()
+	}
+}
+
+func (cont *AciController) SubnetDeleted(dn string) {
+	cont.log.Debug("Before deleting dn: ", dn, " CachedSubnets.subnetDns Map:  ", cont.apicConn.CachedSubnetDns)
+	
+	if _, ok := cont.apicConn.CachedSubnetDns[dn]; ok {
+		delete(cont.apicConn.CachedSubnetDns, dn)
+	} else {
+		cont.log.Debug("This shouldn't happen!")
+	}
+	cont.log.Debug("After Delete: CachedSubnets.subnetDns Map:  ", cont.apicConn.CachedSubnetDns)
+	cont.scheduleRdConfig()
+}
+
+func (cont *AciController) triggerSyncRdConfig() {
+	cont.log.Debug("DummySync!")
+}
+	
+func (cont *AciController) UpdateSubnetDnCache(subnetDn string, subnetIp string, aciVrfDn string) {
+	cont.log.Debug("aciVrfDn: ",aciVrfDn, "; Processing SubnetDn: ", subnetDn)
+	subnetDelimiter := "/subnet"
+	subnetParentDn := strings.Split(subnetDn, subnetDelimiter)[0]
+	var inCache = false
+	for k,_ := range cont.apicConn.CachedSubnetDns {
+		if subnetParentDn == strings.Split(k, subnetDelimiter)[0] {
+			cont.log.Debug("subnetParentDn is in cachedSubnets:", subnetParentDn)
+			cont.apicConn.CachedSubnetDns[subnetDn] = subnetIp
+			inCache = true
+			cont.log.Debug("cachedSubnetDns Map:  ", cont.apicConn.CachedSubnetDns)
+			continue
+		}
+	}
+	if !inCache {
+		cont.BuildSubnetDnCache(subnetDn, aciVrfDn)
+		cont.triggerSyncRdConfig()
+	}
+}
+
+func (cont *AciController) BuildSubnetDnCache(dn string, aciVrfDn string) {
+	cont.log.Debug("aciVrfDn: ",aciVrfDn, "; Processing dn: ", dn)
+	var vrfBdDns  []string
+	var vrfEpgDns []string
+
+	// Get all the BDs related to aciVrf and then all EPGs related to those BDs
+	bdRsDelimiter := "/rsctx"
+	epgRsDelimiter := "/rsbd"
+	subnetDelimiter := "/subnet"
+	bdFilter := fmt.Sprintf("query-target-filter=and(wcard(fvRsCtx.tDn,\"%s\"))",aciVrfDn)
+	bdArgs := []string{
+		"rsp-prop-include=config-only",
+		"query-target=subtree",
+		bdFilter,
+	}
+
+	epgArgs := []string{
+		"query-target=subtree",
+		"target-subtree-class=fvRsBd",
+	}
+
+	subnetArgs := []string{
+		"rsp-prop-include=config-only",
+	}
+
+	bdUri := fmt.Sprintf("/api/node/class/fvBD.json?%s",strings.Join(bdArgs, "&"))
+	epgUri := fmt.Sprintf("/api/node/class/fvAEPg.json?%s",strings.Join(epgArgs, "&"))
+	SubnetUri := fmt.Sprintf("/api/node/class/fvSubnet.json?%s",strings.Join(subnetArgs, "&"))
+
+    apicresp, err := cont.apicConn.GetApicResponse(bdUri)
+	if err!= nil {
+		return
+	}
+    for _, obj := range apicresp.Imdata {
+        for _, body := range obj {
+			dn,ok := body.Attributes["dn"].(string)
+			if !ok {
+				continue
+			}
+			vrfBdDns = append(vrfBdDns, strings.Split(dn, bdRsDelimiter)[0] )
+		}
+    }
+    sort.Strings(vrfBdDns)	
+	cont.log.Debug("aciVrfBdDns: ", vrfBdDns)
```

Comment:
nit: check indentation here

Reply:
ran go fmt; you will see many other indentation changes now.